### ISSUE TITLE
IIFEs should not be CFA-inlined if the FE is a generator function

### ISSUE THREAD

**Issue by RyanCavanaugh** [microsoft/TypeScript#30907]
This is an incorrect error because generators don't run on invocation

```ts
function wrapI() {
    const iter = (function* foo() {
        iter; // use before def error
        yield 1;
    })();
}
```

cc @bterlson

**Comment by bterlson** [comment]
Note that this also applies to async generators which have similar semantics when immediately invoked.

**Comment by DanielRosenwasser** [comment]
Weird because #23565 was already fixed?

**Comment by ajafff** [comment]
The use before define check doesn't use the control flow graph
